torch/package/mangling.md

```diff
@@ -0,0 +1,101 @@
+# Import mangling in `torch.package`
+
+## Mangling rules
+These are the core invariants; if you are changing mangling code please preserve them.
+
+1. For every module imported by `PackageImporter`, two attributes are mangled:
+    - `__module__`
+    - `__file__`
+2. No mangled names should be serialized by `PackageExporter`.
+3. Every entry point to `PackageImporter` and `PackageExporter` must
```

Comment:
I feel like this rule should be that you must pass demanded names to the package importer and exporter. The surface area of the importer/exporter API is to large to try to maintain this invariant.